### PR TITLE
feat: refresh token and tests

### PR DESCRIPTION
**Pull request by samucorea**
# Description

Adds Unit Test project, and Refresh Token functionality.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: NoNicotine_Data/NoNicotine_Data.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
   <ItemGroup>
     <Folder Include="Context\" />
     <Folder Include="DTO\" />
+    <Folder Include="Migrations\" />
```
Comment (gasimon12) [inline NoNicotine_Data/NoNicotine_Data.csproj:12]: What's the reason behind this change?

File: NoNicotineAPI.UnitTests/Queries/CreatePatientCommandHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+//            result.Succeeded.ShouldBeTrue();
+//            result.Data.ShouldNotBeNull();
+//            result.Data.Name.ShouldBe("Samuel Garcia");
+
+//            patients.Count.ShouldBe(5);
+
+
+//        }
+//    }
+//}
```
Comment (gasimon12) [inline NoNicotineAPI.UnitTests/Queries/CreatePatientCommandHandlerTests.cs:89]: Why did you comment all the lines in this file?

File: NoNicotineAPI.UnitTests/UnitTest1.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+{
+    public class UnitTest1
+    {
+        [Fact]
+        public void Test1()
+        {
+
+        }
+    }
+}
```
Comment (gasimon12) [inline NoNicotineAPI.UnitTests/UnitTest1.cs:11]: Pls delete this file

File: NoNicotine_Business/Handler/CreatePatientCommandHandler.cs [hunk 4]
```diff
@@ -5,6 +5,7 @@
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 using NoNicotine_Business.Commands;
+using NoNicotine_Business.Repositories;
```
Comment (gasimon12) [inline NoNicotine_Business/Handler/CreatePatientCommandHandler.cs:8]: All of this changes shouldn't be merged with the master branch (they're exclusively for testing purposes)

File: NoNicotine_Business/Handler/GetPatientQueryHandler.cs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ﻿using MediatR;
 using Microsoft.Extensions.Logging;
 using NoNicotine_Business.Queries;
+using NoNicotine_Business.Repositories;
```
Comment (gasimon12) [inline NoNicotine_Business/Handler/GetPatientQueryHandler.cs:4]: Changes exclusively for testing purposes shouldn't be merged with the master branch

**Review by gasimon12 — CHANGES_REQUESTED** [review]

File: NoNicotine_Business/Handler/GetPatientQueryHandler.cs [hunk 6]
```diff
@@ -16,12 +17,10 @@ namespace NoNicotine_Business.Handler
     public class GetPatientQueryHandler : IRequestHandler<GetPatientQuery, Response<Patient>>
     {
 
-        private readonly AppDbContext _context;
-        private readonly ILogger<GetPatientQueryHandler> _logger;
```
Comment (gasimon12) [inline NoNicotine_Business/Handler/GetPatientQueryHandler.cs:20]: Why did you remove the logger interface?

Comment (samucorea) [reply NoNicotine_Business/Handler/GetPatientQueryHandler.cs:20]: Because we were not using it... jeje

Comment (gasimon12) [reply NoNicotine_Business/Handler/GetPatientQueryHandler.cs:20]: But will we never use it? xd

File: NoNicotine_Data/Context/AppDbContext.cs [hunk 7]
```diff
@@ -18,12 +18,14 @@ public AppDbContext(DbContextOptions<AppDbContext> options) : base(options)
             
         }
 
+        public AppDbContext() { }
```
Comment (gasimon12) [inline NoNicotine_Data/Context/AppDbContext.cs:21]: What is this for?

Comment (samucorea) [reply NoNicotine_Data/Context/AppDbContext.cs:21]: it was me aimlessly trying to mock db context. but it has no use

File: NoNicotine_Business/Handler/GetPatientQueryHandler.cs (hunk 6 above)
Comment (samucorea) [reply NoNicotine_Business/Handler/GetPatientQueryHandler.cs:20]: maybe later, but for now nop

**Review by gasimon12 — APPROVED** [review]
